src/opm/output/eclipse/Summary.cpp

```diff
@@ -1468,11 +1468,19 @@ void Summary::add_timestep( int report_step,
         }
     }
 
-    for (const auto& pair: st) {
-        const auto* key = pair.first.c_str();
+    {
+        const ecl_sum_type * ecl_sum = this->ecl_sum.get();
+        const ecl_smspec_type * smspec = ecl_sum_get_smspec(ecl_sum);
+        for (int node_index = 0; node_index < ecl_smspec_num_nodes( smspec ); node_index++) {
```

Comment:
I suppose `ecl_smspec_num_nodes()` is constant and **might** be pulled out of the loop&mdash;especially if its complexity is not constant time.

Reply:
The function `ecl_smspec_num_nodes( )` takes a pointer to `const`as argument, and the argument variable `smspec`*is* pointer to `const` - so I would really assume this function was only evaluated once anyways? Whatever - I will pull it out.